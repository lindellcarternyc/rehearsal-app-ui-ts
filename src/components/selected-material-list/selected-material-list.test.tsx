import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as enzyme from 'enzyme'
import * as adapter from 'enzyme-adapter-react-16'
enzyme.configure({adapter: new adapter()})

import SelectedMaterialList, { SelectedMaterialListProps } from './selected-material-list'
import SelectedMaterialListItem from './selected-material-list-item'

describe('SelectedMaterialList', () => {
  const material = [
    'test 1', 'test 2', 'test 3'
  ]
  const props: SelectedMaterialListProps = {
    material
  }
  it('should render a list with a removeCallback', () => {
    let _material = material
    const callback = (id: number) => {
      _material.slice(id, 1)
    }
    const _props: SelectedMaterialListProps = {...props, removeMaterial: callback}
    const list = renderer.create(
      <SelectedMaterialList {..._props} />
    )

    const tree = list.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render a list without a callback', () => {
    const tree = renderer.create(
      <SelectedMaterialList {...props} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders the correct number of list items', () => {
    const list = enzyme.shallow(
      <SelectedMaterialList {...props}/>
    )
    expect(list.find(SelectedMaterialListItem)).toHaveLength(3)
  })
})