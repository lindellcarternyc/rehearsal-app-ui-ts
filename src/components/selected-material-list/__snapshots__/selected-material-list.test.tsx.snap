// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedMaterialList should render a list with a removeCallback 1`] = `
<div
  style={
    Object {
      "marginBottom": "1rem",
    }
  }
>
  <h4
    className="ui header"
  >
    Material
  </h4>
  <div
    className="ui bulleted list"
    role="list"
  >
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test 1
       
      <i
        aria-hidden="true"
        className="cancel icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test 2
       
      <i
        aria-hidden="true"
        className="cancel icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test 3
       
      <i
        aria-hidden="true"
        className="cancel icon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`SelectedMaterialList should render a list without a callback 1`] = `
<div
  style={
    Object {
      "marginBottom": "1rem",
    }
  }
>
  <h4
    className="ui header"
  >
    Material
  </h4>
  <div
    className="ui bulleted list"
    role="list"
  >
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test 1
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test 2
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      test 3
    </div>
  </div>
</div>
`;
